### DISCUSSION TITLE
Algoritmekader v1.1

### DISCUSSION THREAD

**Discussion by ruthkoole** [MinBZK/Algoritmekader#81] [Announcements]
Dit is een doorontwikkeling op het [Implementatiekader ‘Verantwoorde inzet van algoritmen’](https://www.rijksoverheid.nl/documenten/rapporten/2023/06/30/implementatiekader-verantwoorde-inzet-van-algoritmen) dat op 7 juli 2023 is verzonden naar de Tweede Kamer als bijlage van de [Kamerbrief algoritmen en AI](https://www.rijksoverheid.nl/documenten/kamerstukken/2023/07/10/kamerbrief-over-algoritmen-reguleren). Naar deze versie wordt verwezen als het Implementatiekader v1.0. Dit Implementatiekader wordt doorontwikkeld, zowel op inhoud als op vorm. 

## Wat is er veranderd?
- Het Implementatiekader ‘Verantwoorde inzet van algoritmen' is in oktober 2023 hernoemd naar 'het Algoritmekader'.
- Hierbij is ook de visie op het product aangepast: van een kader in de vorm van één document naar een interactieve kennisbank.
- Het samenspel van vereisten en maatregelen is duidelijker gemaakt: de eerste set van vereisten is al toegevoegd.
- Het Algoritmekader is interactief doorzoekbaar gemaakt: dat kan vooralsnog vanuit de fasen van de levenscyclus van een algoritme en de bouwblokken. Dit moet het mogelijk maken voor gebruikers om op een interactieve manier door de informatie te navigeren. 
- Algemene teksten zijn toegevoegd over: het algoritmekader in het algemeen, de levenscyclus, de vereisten, de maatregelen, de rollen en de instrumenten. 

<hr /><em>This discussion was created from the release <a href='https://github.com/MinBZK/Algoritmekader/releases/tag/v1.1'>Algoritmekader v1.1</a>.</em>

**Comment by HaitskevdLinde** [comment]
Mooie mijlpaal. Ik ben heel benieuwd naar de reacties/feedback.

**Comment by floort** [comment]
N.a.v de [instructie](https://algoritmes.pleio.nl/news/view/a7efd9be-a564-49ef-91c5-5f8ad080e96c/1e-release-algoritmekader) op Pleio haal ik de discussie hier naartoe. In PR #71 heb ik een voorstel gedaan om de verplichtingen onder artikel 5 AVG samen te vatten zodat de overige technieken en regels een duidelijker kader hebben. De PR is ruim voor de release ingediend, niet meegenomen en nu weer bijgewerkt voor de nieuwe release. 

Procesmatig zou ik voorstellen dat het handig is om pull request tijdens het werk aan de release te beoordelen, van feedback te voorzien en/of over te nemen. Volgens mij is dat in lijn met de contributing guidelines uit commit 5323c245271568676c5674343c0e195c6a29c7f5

Inhoudelijk is er wel een verwijzing naar artikel 5 AVG meegenomen in commit 5323c245271568676c5674343c0e195c6a29c7f5.
Deze verwijzing is echter onvolledig (5(1) onbreekt, maar bevat juist cruciale details) en onjuist.  Zo staat er vermeld dat de verantwoordingsplicht niet van toepassing is op niet-impactvolle algoritmen terwijl artikel 5, net als alle andere in het document genoemde artikelen, van toepassing is op alle verwerkingen waar de AVG op van toepassing is. 

Mijn zorg is dat de impact van artikel 5 AVG niet duidelijk gecommuniceerd wordt. Artikel 5 draait voor een groot deel de bewijslast om. Je moet kunnen aantonen dat een algortime (waar persoonsgegevens worden verwerkt) rechtmatig wordt ingezet en het is niet genoeg om best-effort transparantie te geven. Dat is voor organisaties die aan de slag gaan zeer relevant, want dat is in de uitvoering een zeer ingrijpend verschil dat nu niet nadrukkelijk wordt uitgelegd. Maar ook voor burgers die de rechtmatigheid willen controleren is het belangrijk waar, primair, de bewijslast ligt en hoe gaten in de onderbouwing geïnterpreteerd moeten worden.

Ik denk dat de keus over de uitleg van artikel 5 AVG een belangrijke keus is voor de toon van het algortimekader. Is het de bedoeling om mensen te informeren over regels, verplichtingen, methoden die voor hen relevant/verplicht (kunnen) zijn of is het kader meer bedoeld als een verzameling vrijblijvende tips waar mensen uit kunnen kiezen wat ze willen? Ik zou voor het eerste kiezen, maar uit de toon van het algoritmekader maak ik op dat er misschien meer iets richting het tweede is gekozen. Misschien is het goed om die keuze te documenteren zodat dat voor lezers én mensen die willen bijdragen duidelijk is.

**Comment by floort** [comment]
@ruthkoole Naar aanleiding van je opmerking tijdens de sprintreview op 11 april (minuut 25/26 van de opname): Het is niet helemaal waar dat alles te volgen is via branches. Er wordt ook gewerkt met branches die verborgen zijn, waardoor de commits slechts gedeeltelijk inzichtelijk zijn en waardoor bijdragen via pull requests niet gebaseerd kunnen worden op de meest recente versie van de documenten. Ik wil dit benadrukken omdat ik merk dat het erg lastig is om fatsoenlijk bij te dragen. Dat kan eigenlijk het beste als kleine opzichzelfstaande veranderingen voor iedereen evenveel toegankelijk zijn en het liefst ook als discussies over waarom keuzes gemaakt worden hier bij bijvoorbeeld de pull requests gevoerd worden zodat iedereen kan meelezen of zelfs argumenten bijdragen. 


Een praktisch voorbeeld van de lastig te volgen commits is bijvoorbeeld hier te vinden: https://github.com/MinBZK/Algoritmekader/pull/69/files/a2bc55831cfb7b4ce4a1dd0c433020d37f1f627a..42f445bd7a80a4cc2472a35707e6ddf7fbf776b1

**Comment by ruthkoole** [comment]
Hoi @floort,

Onze werkgroep rondom privacy en gegevensbescherming gaat inhoudelijk kijken naar jouw pull-request rondom artikel 5 van de AVG. Dit proberen we te verwerken voor de volgende release. Verwoording maakt inderdaad uit, dus we kijken er graag naar met een groep van experts. 

Ik ben het met jou eens dat het handig is om een pull-request tijdens het werken aan de release te beoordelen. Dat is ook de werkwijze die we voor ons zien en hoe we het beschreven hebben in de contributing guidelines. Afhankelijk van het onderwerp en de aard van de pull-request kan het soms wat langer duren om dit te reviewen (we pakken de onderwerpen één voor één op, dus het soms het is het logischer een inhoudelijke suggestie door te schuiven naar een volgende release). 

Het werken met de releases is nieuw voor ons, waardoor de aanloop naar de 1e release niet precies is verlopen zoals bedoeld. We proberen daar strakker in te zijn vanaf nu, en hopen dat het door het werken met de releases ook duidelijker is aan welke onderwerpen er gewerkt wordt. 

Wat betreft jouw opmerking op de branches, snap ik dat het voorheen (en bij het voorbeeld waar je naar verwijst) lastig te volgen is geweest. Dat is in aanloop naar release v1.1 niet duidelijk genoeg geweest. Nu de structuur van de website iets beter staat, en we werken met releases moet het volgens mij makkelijker te volgen zijn. Een voorbeeld van hoe we dit beter proberen te doen, is dat we een milestone met onderliggende issues hebben gemaakt voor de volgende release, zie https://github.com/MinBZK/Algoritmekader/milestone/1. We proberen de pull-requests kleiner te maken zodat het makkelijker te volgen is. Dat er voorheen is gewerkt met verborgen branches is overigens niet waar. Ik hoop dat dit in het vervolg beter gaat, en dat het makkelijker wordt om bij te dragen. 

We hebben ervoor gekozen om voor bepaalde onderwerpen met werkgroepen te werken aan de inhoud. De inhoudelijke discussie vindt vaak plaats in die werkgroep. Het is helaas niet realistisch om alle onderbouwingen en afwegingen in detail ook op Github te plaatsen. Ik snap dat dat fijn zou zijn voor de lezer, maar dat is helaas met onze beschikbare resources niet heel realistisch. We proberen de pull-requests wel beter toe te lichten. Daarnaast is het natuurlijk ook mogelijk om deel te nemen aan één van de werkgroepen.

**Reply by floort** [reply]
Bedankt voor de uitleg.
Ik heb het misschien niet goed benoemd, maar ik kan de gehele 'templates' branch niet meer vinden en dat maakt het lastiger om te achterhalen hoe bijdragen zijn geleverd. Misschien zie ik dan iets over het hoofd, maar voor mij is het verborgen. Ik hoor graag waar die branch wel te vinden is. Dat zou kunnen helpen bij het volgen van de redeneringen achter de logica.

Wat misschien helpt om de waarde van heldere en specifieke commits te illustreren is door via de 'git blame' functionaliteit naar de documenten te kijken. Je zou voor elk stukje tekst moeten kunne zien wie dat voor het laatst heeft 'aangeraakt' en wat de reden was voor de laatste wijziging die dat stukje tekst heeft ondergaan. Verdwenen branches en grote bulk commits maken die reconstructie onnodig lastig.

Het combineren van twee (of meer) samenwerkingsvormen is lastig. Het is wel belangrijk om de tijd van mensen die vrijwillig bijdragen te respecteren door zo goed mogelijk de kennis te delen. Een heads-up voor specifieke onderdelen die tijdelijk even 'bezet' zijn door een werkgroep die (voor buitenstaanders) even onzichtbaar aan het werk zijn en het zo kort mogelijk houden van die blokken kan het makkelijker maken om deze twee werkwijzen naast elkaar te laten bestaan.

**Reply by ruthkoole** [reply]
De templates branch is inmiddels inderdaad verwijderd. Er heeft nooit een merge plaatsgevonden van de template branch naar main. We hebben deze branch gebruikt om dingen uit te proberen voor een nieuwe structuur van het algoritmekader. Uiteindelijk hebben we via pull-request https://github.com/MinBZK/Algoritmekader/pull/77 de wijzigingen die we wilden doen gedaan. Deze pull request is wel gemergd. Juist omdat de templates branch zo onoverzichtelijk was geworden, en we daar ook wat dingen uit hebben geprobeerd die uiteindelijk niet wenselijk leken te zijn, leek het beter een nieuwe branch te maken met de uiteindelijke wijzigingen. Maar ik ben het met jou eens dat die werkwijze niet optimaal was. 

Eens dat de verschillende werkvormen op sommige vlakken lastig te combineren zijn. We zoeken naar de meest optimale weg om dat te combineren. Hopelijk is het met behulp van de milestones duidelijker waar er op dit moment aan gewerkt wordt.

**Reply by floort** [reply]
De reden achter het verdwijnen van de 'templates' branch is ongeveer wat ik had verwacht. Wat mij betreft is het ook niet nodig om te verantwoorden waarom het zo is gegaan. Maar ik wil wel duidelijk maken dat het wissen van geschiedenis het lastiger maakt om te volgen wat er gebeurt en om bij te dragen in de hoop dat er een volgende keer anders kan.

**Comment by sjoerdly** [comment]
Hi! Geen inhoudelijk commentaar, maar een praktische tip. Wellicht is het handig om in het vervolg in de release notes ook linken naar de daadwerkelijke kader website: https://minbzk.github.io/Algoritmekader/. En eventueel zelfs deeplinken aan de hand van de wijzigingen. 

Ook is het handig om ergens op https://minbzk.github.io/Algoritmekader/
een link te plaatsen naar de nieuwsbrief als je bv nog niet direct wilt bijdragen maar wel op de hoogte wilt worden gehouden van ontwikkelingen https://algoritmeregister.email-provider.eu/memberforms/subscribe/standalone/form/?a=1pjwwoyxrs&l=vdfr1sbovb

**Reply by ruthkoole** [reply]
Goede tips! Dank!  We voegen een link naar de website toe in de volgende release notes, en de bijbehorende discussie. Ook voegen we bij de volgende release een link naar inschrijven voor de nieuwsbrief toe op de homepagina.
